src/config/mod.rs

```diff
@@ -930,7 +981,12 @@ impl Default for LockCfConfig {
 }
 
 impl LockCfConfig {
-    pub fn build_opt(&self, shared: &CfResources, for_engine: EngineType) -> RocksCfOptions {
+    pub fn build_opt(
+        &self,
+        shared: &CfResources,
+        filter_factory: Option<&RangeCompactionFilterFactory>,
```

Comment:
Maybe should be put into `CfResources`.

Reply:
`CfResources` is supposed to be shared between tablets. Stuff like `region_info_accessor` can probably be moved into it though.